Modules/socketmodule.c

```diff
@@ -1877,6 +1916,7 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
     case AF_BLUETOOTH:
     {
         switch (s->sock_proto) {
+#ifndef MS_WINDOWS
```

Comment:
Can we test whether macros are defined rather than checking for platform? (Also, is `BTHPROTO_L2CAP` the same as `BTPROTO_L2CAP` here?)

Reply:
I changed all of the `ifndef MS_WINDOWS` tests to `ifdef BTPROTO_X` and added/changed such tests in other cases (e.g. `ifdef BTPROTO_SCO` instead of checking for not FreeBSD).

`BTHPROTO_L2CAP` is the same as `BTPROTO_L2CAP`. Windows defines the constant, but winsock does not implement L2CAP support. I was able to build with this, but creating a socket with `BTPROTO_L2CAP` with winsock fails.